Fix Travis-CI for Windows

Travis-CI is failing in the cross-compilation of Windows binaries. I believe the error is here:

```
make[2]: Entering directory '/home/travis/build/spl/teckit/windows-build32/lib'
  CXX      ../source/Compiler.lo
[...]
i686-w64-mingw32-windres   -o Compiler_ver.o ../../source/Compiler_ver.rc
  CXXLD    TECkit_Compiler_x86.la
/usr/bin/ld: unrecognized option '--add-stdcall-alias'
/usr/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
Makefile:687: recipe for target 'TECkit_Compiler_x86.la' failed
[...]
The command "./build-windows-binaries.sh" exited with 2.
```

I looked into the `ld` error on Travis-CI:

```sh
$ /usr/bin/ld --version
GNU ld (GNU Binutils for Ubuntu) 2.26.1
```

```sh
$ /usr/bin/ld --help
Usage: /usr/bin/ld [options] file...
Options:
[...]
  --add-stdcall-alias                Export symbols with and without @nn
[...]
```

I'm not sure what the error means, since it appears that `ld` does have the `--add-stdcall-alias` flag.

But is this the right `ld` executable here? Or is there a different one from the MinGW packages that should be used?

In the process of looking into this, I found this warning from `../configure`:

```
configure: WARNING: using cross tools not prefixed with host triplet
```

Searching for the warning led me to a page in the `autoconf` manual on [Specifying target triplets](https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Specifying-Target-Triplets.html). I wonder if that could be related to the `ld` error.

There are a number of posts around the internet with similar `ld` error reports, but the solution never seems to have anything to do with the `ld` flag `--add-stdcall-alias` itself. Plus, I'm assuming the TECkit developers use this script locally, so it must work in some places.

--- a/.travis.yml
+++ b/.travis.yml
@@ -38,9 +38,6 @@ addons:
       - debhelper
       - cdbs
       # For cross-compiling Windows executables
-      - gcc-mingw-w64-i686
-      - g++-mingw-w64-i686
-      - binutils-mingw-w64-i686
-      - gcc-mingw-w64-x86-64
-      - g++-mingw-w64-x86-64
-      - binutils-mingw-w64-x86-64
+      - mingw-w64
+      - dos2unix
+      - groff

--- a/build-windows-binaries.sh
+++ b/build-windows-binaries.sh
@@ -22,7 +22,7 @@ do
     BUILD=$(../config.guess)
     set_compiler $bit
 
-    echo "build is +${BUILD}+ host is +${HOST}+ in binaries with bit +${bit}+"
+    echo "BUILD=${BUILD} HOST=${HOST} bit=${bit}"
     ../configure --build=$BUILD --host=$HOST --with-old-lib-names --without-system-zlib --enable-final --disable-dependency-tracking
     make
     make install-strip DESTDIR=`pwd`/inst

--- a/build-windows-common
+++ b/build-windows-common
@@ -7,7 +7,9 @@ find_compiler()
     # Check the various names used for mingw
     for PREFIX in "$@"
     do
-        if which $PREFIX-gcc >/dev/null
+        CC=$PREFIX-gcc
+        CXX=$PREFIX-g++
+        if which $CC > /dev/null && which $CXX > /dev/null
         then
             HOST=$PREFIX
             break

